### PR TITLE
fix saving votes from inside a card

### PR DESCRIPTION
**Pull request by mattcasey**
I noticed I couldn't save a new vote when looking at a card view inside of focalboard. It's because the pageId was 'null' inside the hook - it doesn't seem to get upated when the URL changes.

I also moved the call to useVotes for `createVote` into the modal component because all the parents were having to do the same logic.

### CODE REVIEW

File: hooks/useVotes.tsx [hunk 1]
```diff
@@ -87,7 +87,6 @@ export function VotesProvider ({ children }: { children: ReactNode }) {
     const extendedVote = await charmClient.createVote({
       ...votePayload,
       createdBy: user.id,
-      pageId: cardId ?? currentPageId,
```
Comment (mattcasey) [inline hooks/useVotes.tsx:90]: The other place we use cardId in this hook is to tell if we should skip loading votes on line 38. Do you know @Devorein what was the context for that logic? Why can't we load votes for any page?

File: components/votes/components/CreateVoteModal.tsx [hunk 2]
```diff
@@ -78,20 +79,21 @@ function InlineVoteOptions (
 
 interface CreateVoteModalProps {
   onClose?: () => void;
-  createVote: (votePayload: Omit<VoteDTO, 'createdBy' | 'spaceId'>) => Promise<ExtendedVote>;
-  postCreateVote?: (vote: ExtendedVote) => void;
+  onCreateVote?: (vote: ExtendedVote) => void;
```
Comment (mattcasey) [inline components/votes/components/CreateVoteModal.tsx:82]: trying to standardize on callback names: they should start with the word 'on'. and it is usually implied that the action already took place

Comment (Devorein) [reply components/votes/components/CreateVoteModal.tsx:82]: Good idea. I will keep that in mind from now 'on' when naming props

File: hooks/useVotes.tsx (hunk 1 above)
Comment (Devorein) [reply hooks/useVotes.tsx:90]: I can't exactly remember the reasoning behind it but I think it had something to do with the right sidebar showing inside card view which we decided on not showing. As for the global context, that would require a separate PR for the refactor I believe, updating currentPageId with the cardId if any card page is opened, right now it points to the board page.

**Review by Devorein — APPROVED** [review]

File: hooks/useVotes.tsx (hunk 1 above)
Comment (Devorein) [reply hooks/useVotes.tsx:90]: I can't exactly remember the reasoning behind it but I think it had something to do with the right sidebar showing inside card view which we decided on not showing. As for the global context, that would require a separate PR for the refactor I believe, updating currentPageId with the cardId if any card page is opened, right now it points to the board page.

Comment (mattcasey) [reply hooks/useVotes.tsx:90]: Oh, ok.. i wonder if we could control that from the card view itself, though, even if we had to pass a prop down thru to DocumentPage. we don't need the cardId, just the fact that the sidebar shouldn't be viewed

**Pull request merged**